DegreeProject/Assets/Scripts/Enemy/FollowState.cs

```diff
@@ -34,13 +34,52 @@ public override void EnterState(AI owner)
     public override void ExitState(AI owner)
     {
         Debug.Log("Exiting FollowState.");
+
+        if (owner.playerObject)
+        {
+            DetectNearbyPlayer(owner);
+        }
     }
 
     public override void UpdateState(AI owner)
     {
-        if (!owner.switchState)
+        if (!owner.playerObject)
+        {
+            owner.stateMachine.ChangeState(PatrolState.stateInstance);
+        }
+
+        if (owner.playerObject) // To ensure that we don't call null reference after playerObject is set to null
+        {
+            FollowPlayer(owner);
+        }
+    }
+
+    private void DetectNearbyPlayer(AI owner)
+    {
+        // When exiting follow state, check if there is any other players
+        // still inside detection circle 
+        owner.detectionCircle = Physics2D.OverlapCircle(owner.transform.position, owner.detectionRadius);
```

Comment:
Med Overlap och raycast så är det viktigt att jobba med rätt physics layers. Kom även ihåg att Overlap och Raycast kan träffa andra objekt, om det finns fler i samma lager som t.ex. spelaren. 
Jag brukar använda OverlapCircleAll/RaycastAll, så jag kan gå igenom alla resultat och vara säker på att det jag letar efter inte finns bland resultaten.

Fun fact! 2D RaycastAll och OverlapCircleAll sorterar sina resultat efter distans, detta gör inte 3Ds counter part metoder. 3Ds metoder returnerar osorterade listor, och det är upp till dig att sortera dom efter distans. The more you know :D